lib/moleculer/transporters/redis.rb

```diff
@@ -70,26 +94,14 @@ def connect
                 @logger.debug "starting subscription to '#{@channel}'"
                 @connection.subscribe(@channel) do |on|
                   on.unsubscribe do
-                    @logger.debug "disconnecting channel '#{@channel}'"
-                    @connection.disconnect!
+                    unsubscribe
                   end
 
                   on.message do |_, message|
-                    if message == @disconnect_hash.value
-                      @connection.unsubscribe
-                    elsif message.split(".")[-1] != "disconnect"
-                      begin
-                        set_disconnect
-                        parsed = @serializer.deserialize(message)
-                        if parsed && parsed["sender"] != Moleculer.node_id
-                          @logger.trace "received message '#{@channel}'", parsed
-                          packet = @packet_type.new(parsed)
-                          @block.call(packet)
-                        end
-                      rescue StandardError => e
-                        @logger.error e
-                      end
-                    end
+                    packet = MessageProcessor.new(self).process(message)
```

Comment:
let's treat this like a service and make the process method behave like a singleton